.changeset/engine-enhancements.md

```diff
@@ -0,0 +1,74 @@
+---
+"thirdweb": minor
+---
+
+Enhanced Engine functionality with search transactions and batch transaction support:
+
+- Added `Engine.searchTransactions()` to search for transactions by various filters (id, chainId, from address, etc.)
+
+  ```ts
+  // Search by transaction IDs
+  const transactions = await Engine.searchTransactions({
+    client,
+    filters: [
+      {
+        field: "id",
+        values: ["1", "2", "3"],
+        operation: "OR",
+      },
+    ],
+  });
+
+  // Search by chain ID and sender address
+  const transactions = await Engine.searchTransactions({
+    client,
+    filters: [
+      {
+        filters: [
+          {
+            field: "from",
+            values: ["0x1234567890123456789012345678901234567890"],
+            operation: "OR",
+          },
+          {
+            field: "chainId",
+            values: ["8453"],
+            operation: "OR",
+          },
+        ],
+        operation: "AND",
+      },
+    ],
+    pageSize: 100,
+    page: 0,
+  });
+  ```
+
+- Added `serverWallet.enqueueBatchTransaction()` to enqueue multiple transactions in a single batch
+
+  ```ts
+  // Prepare multiple transactions
+  const transaction1 = claimTo({
+    contract,
+    to: firstRecipient,
+    quantity: 1n,
+  });
+  const transaction2 = claimTo({
+    contract,
+    to: secondRecipient,
+    quantity: 1n,
+  });
+
+  // Enqueue as a batch
+  const { transactionId } = await serverWallet.enqueueBatchTransaction({
+    transactions: [transaction1, transaction2],
+  });
+
+  // Wait for batch completion
+  const { transactionHash } = await Engine.waitForTransactionHash({
```

Comment:
if I pass multiple transactions which transaction hash am I getting back? or should I get back an array?

Reply:
Started with an array, but it actually will batch into a single transaction so you only get one id back. There might be a case where you get multiple ones if using eoa, but you probably wouldn't want to batch for eoas anyways?

So found it more intuitive to return a single id so you understand it's a single tx, just like semdBatchTransaction